ipa-core/src/report/hybrid.rs

```diff
@@ -22,6 +83,114 @@ where
     breakdown_key: Replicated<BK>,
 }
 
+impl<BK: SharedValue> Serializable for HybridImpressionReport<BK>
+where
+    BK: SharedValue,
+    Replicated<BK>: Serializable,
+    <Replicated<BK> as Serializable>::Size: Add<U16>,
+    <<Replicated<BK> as Serializable>::Size as Add<<Replicated<BA64> as Serializable>::Size>>:: Output: ArrayLength,
+{
+    type Size = <<Replicated<BK> as Serializable>::Size as Add<<Replicated<BA64> as Serializable>::Size>>:: Output;
+    type DeserializationError = InvalidHybridReportError; //<Replicated<BK> as Serializable>::DeserializationError;
+
+    fn serialize(&self, buf: &mut GenericArray<u8, Self::Size>) {
+        let mk_sz = <Replicated<BA64> as Serializable>::Size::USIZE;
+        let bk_sz = <Replicated<BK> as Serializable>::Size::USIZE;
+
+        self.match_key
+            .serialize(GenericArray::from_mut_slice(&mut buf[..mk_sz]));
+
+        self.breakdown_key
+            .serialize(GenericArray::from_mut_slice(&mut buf[mk_sz..mk_sz + bk_sz]));
+    }
+    fn deserialize(buf: &GenericArray<u8, Self::Size>) -> Result<Self, Self::DeserializationError> {
+        let mk_sz = <Replicated<BA64> as Serializable>::Size::USIZE;
+        let bk_sz = <Replicated<BK> as Serializable>::Size::USIZE;
+        let match_key =
+            Replicated::<BA64>::deserialize(GenericArray::from_slice(&buf[..mk_sz]))
+            .map_err(|e| InvalidHybridReportError::DeserializationError("match_key", e.into()))?;
```

Comment:
is there a way to show more context in this error? Like where did it happen?

I'm imagining we get this error and we need to find the issue

Reply:
<img width="925" alt="Screenshot 2024-10-16 at 2 41 40 PM" src="https://github.com/user-attachments/assets/b95cad6d-eb38-4e26-9b88-8942df2fbe8d">
Rust-analyzer indicates that this operation is infallible. I would guess that's because it looks like this can take arbitrary values, but then why isn't `Replicated::<BK>::deserialize` infallible?

Reply:
>  why isn't Replicated::< BK >::deserialize infallible
two reasons.
* We use prime fields where values > PRIME are rejected at deserialization layer to avoid silent truncation
* We use bitstrings like BA3, BA5 where we also reject values greater than MAX

Our macro implementation guarantees that Replicated<BA8> is infallible, but generic `Replicated<S>` is not (unless a trait bound is added here)

For Replicated::<BA64> I think we could use `deserialize_infallible` method that should be available

Reply:
I didn't see an easy way to use `deserialize_infallible` so I went with `unwrap_infallible`

Reply:
Should be relatively straightforward

```rust
        let match_key =
            Replicated::<BA64>::deserialize_infallible(GenericArray::from_slice(&buf[..mk_sz]));
```

it seems to compile for me

Reply:
oh hmm, I thought I'd have to implement it or something because rust-analyzer said it didn't recognize `deserialize_infallible`, but I guess it was just slow and I should have tried compiling lol